### PR TITLE
docs: update contract walkthru

### PR DESCRIPTION
**Pull request by mujahidkay**
Closes: https://github.com/Agoric/documentation/issues/1173

### Description

This PR:

 - [x] Updates contract listing code to use latest version
 - [x] Updates alice.js accordingly
 - [x] Updates mermaid diagrams
 - [x] Updates contract-walkthu in accordance with latest example

### CODE REVIEW

**Comment by mujahidkay** [comment]
@amessbee could you please take a look at this? :) Would appreciate any feedback.

File: snippets/zoe/src/offer-up.contract.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  return entries.map(([_k, ct]) => ct);
+};
+// #endregion bagUtils
+
+/**
+ * In addition to the standard `issuers` and `brands` terms,
+ * this contract is parameterized by terms for price and,
+ * optionally, a maximum number of items sold for that price (default: 3).
+ *
+ * @typedef {{
```
Comment (dckc) [inline snippets/zoe/src/offer-up.contract.js:32]: I wonder why GitHub actions is issuing a warning here:

```
Syntax error in type: {tradePrice: Amount;maxItems?: bigint;}
```

I wonder if some `tsc` version is out of date somewhere.

I'm not too worried about it just now, though.

**Review by dckc — COMMENTED** [review]
excellent work!

any particular reason why it's still DRAFT?
Also, it looks like it closes the issue.

File: main/guides/zoe/assets/trade-offer-safety-3.mmd [hunk 2]
```diff
@@ -26,4 +26,4 @@ sequenceDiagram
 
     Zoe--)Zoe: escrow Price pmt
 
-    Zoe--)-C: joinHandler(gameSeat)
+    Zoe--)-C: tradeHandler(proceeds)
```
Comment (dckc) [inline main/guides/zoe/assets/trade-offer-safety-3.mmd:29]: looks like `proceeds` got mixed up with `buyerSeat`.

```suggestion
    Zoe--)-C: tradeHandler(buyerSeat)
```

**Review by dckc — CHANGES_REQUESTED** [review]
ah. found one tweak

File: main/guides/zoe/assets/trade-offer-safety-3.mmd (hunk 2 above)
Comment (mujahidkay) [reply main/guides/zoe/assets/trade-offer-safety-3.mmd:29]: `gameSeat` is being referred to as `proceeds` in the current version of contract (the seat created via `makeEmptySeatKit`). Or am I interpreting it wrong? Or was it a typo in the previous iteration of the diagram?

**Comment by mujahidkay** [comment]
> any particular reason why it's still DRAFT?

_Do a Draft PR and have it reviewed by others_ (as per [CONTRIBUTING.md](https://github.com/Agoric/documentation/blob/main/CONTRIBUTING.md)).

File: main/guides/zoe/assets/trade-offer-safety-3.mmd (hunk 2 above)
Comment (mujahidkay) [reply main/guides/zoe/assets/trade-offer-safety-3.mmd:29]: @dckc PTAL at this^

**Comment by dckc** [comment]
> > any particular reason why it's still DRAFT?
> 
> _Do a Draft PR and have it reviewed by others_ (as per [CONTRIBUTING.md](https://github.com/Agoric/documentation/blob/main/CONTRIBUTING.md)).

interesting. you're very thorough indeed. I glazed over that part. I operate under the flip side of...

> If the PR cannot be approved as is (because there's known issues) then it must be a draft PR.

I/we should probably change CONTRIBUTING.md. In most of the shop, we don't necessarily do a DRAFT. In fact, IME the norm is to not invite review until it's non-DRAFT.

File: main/guides/zoe/assets/trade-offer-safety-3.mmd (hunk 2 above)
Comment (dckc) [reply main/guides/zoe/assets/trade-offer-safety-3.mmd:29]: > was it a typo in the previous iteration of the diagram?

yes, I suppose it was.

Comment (mujahidkay) [reply main/guides/zoe/assets/trade-offer-safety-3.mmd:29]: Cool. Then I'll fix this for trade-offer-safety-4 as well. Thanks!

**Review by dckc — APPROVED** [review]
nice work!

**Pull request merged**